### PR TITLE
BaSP-MR-155: Trainer Server

### PR DESCRIPTION
**Pull request by IgnacioGalceran**
Changed functions in controller trainers, creating new trainer now creates a new user in firebase and mongodb. 
Changes in update controller function

### CODE REVIEW

File: src/models/Trainer.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 const mongoose = require('mongoose');
 
 const trainerSchema = new mongoose.Schema({
+  firebaseUid: {
```
Comment (santimenendez98) [inline src/models/Trainer.js:4]: In this part to maintain consistency among all should be "fireBaseUid"

**Review by santimenendez98 — CHANGES_REQUESTED** [review]

File: src/routes/index.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 const express = require('express');
+const { default: verifyToken } = require('../middlewares/authMiddleware');
```
Comment (FontanR) [inline src/routes/index.js:2]: take out this too, please

File: src/routes/index.js [hunk 3]
```diff
@@ -10,7 +11,7 @@ const classes = require('./class');
 const router = express.Router();
 
 router.use('/subscription', subscriptions);
-router.use('/trainer', trainers);
+router.use('/trainer', verifyToken, trainers);
```
Comment (FontanR) [inline src/routes/index.js:14]: we agreed that routes will be implement in a separate ticket

**Review by santimenendez98 — APPROVED** [review]

**Review by FontanR — APPROVED** [review]
LGTM

**Pull request merged**